ethereum/eip712/eip712.go

```diff
@@ -207,8 +207,8 @@ func traverseFields(
 			}
 		}
 
-		// If its a nil pointer, do not include in types
-		if fieldType.Kind() == reflect.Ptr && field.IsNil() {
+		// If field is an empty value, do not include in types, since it will not be present in the object
+		if field.IsZero() {
```

Comment:
> We pushed a “fix” earlier that addresses this—it omits empty values from type generation. It’s a fix in quotes because it causes new issues—it rejects empty values that are not omitted as well. 

@austinchandra perhaps using `IsOmittableField(field)` (see below) instead would resolve the problem you mentioned? 

```go
// IsOmittableField reports whether v is an omittable value for its type.
// It panics if the argument is invalid.
func IsOmittableField(v reflect.Value) bool {
	switch v.Kind() {
	case reflect.Array:
		for i := 0; i < v.Len(); i++ {
			if !v.Index(i).IsZero() {
				return false
			}
		}
		return true
	case reflect.Interface, reflect.Map, reflect.Pointer, reflect.Slice:
		return v.IsNil()
	case reflect.String:
		return v.Len() == 0
	case reflect.Struct:
		for i := 0; i < v.NumField(); i++ {
			if !v.Field(i).IsZero() {
				return false
			}
		}
		return true
	default:
		return false
	}
}
```

Reply:
Ah I see you went with an alternate approach 

https://github.com/evmos/ethermint/blob/16daa44abb17f8215d75c758542d2475d1e0f2b6/ethereum/eip712/eip712.go#L338-L340

does this code work?